pkg/cluster/cluster.go

```diff
@@ -67,3 +75,99 @@ func (ln *localNode) Observe(ckit.Observer) {
 func (ln *localNode) Peers() []peer.Peer {
 	return []peer.Peer{ln.self}
 }
+
+func (ln *localNode) Handler() (string, http.Handler) {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, _ = w.Write([]byte("clustering is disabled"))
+		w.WriteHeader(http.StatusBadRequest)
+	}))
+
+	return "/api/v1/ckit/transport/", mux
+}
+
+// Clusterer implements the behavior required for operating Flow controllers
+// in a distributed fashion.
+type Clusterer struct {
+	Node Node
+}
+
+// New creates a Clusterer.
+func New(log log.Logger, clusterEnabled bool, addr, joinAddr string) (*Clusterer, error) {
+	// Standalone node.
+	if !clusterEnabled {
+		return &Clusterer{Node: NewLocalNode(addr)}, nil
+	}
+
+	host, portStr, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, err
+	}
+	port, err := strconv.Atoi(portStr)
+	if err != nil {
+		return nil, err
+	}
+
+	gossipConfig := DefaultGossipConfig
+	gossipConfig.NodeName = uuid.NewString()
+	gossipConfig.AdvertiseAddr = host
+	err = gossipConfig.ApplyDefaults(port)
+	if err != nil {
+		return nil, err
+	}
+
+	if joinAddr != "" {
+		gossipConfig.JoinPeers = strings.Split(joinAddr, ",")
+	}
+
+	cli := &http.Client{
+		Transport: &http2.Transport{
+			AllowHTTP: true,
+			DialTLS: func(network, addr string, _ *tls.Config) (net.Conn, error) {
+				return net.Dial(network, addr)
+			},
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
```

Comment:
This `InsecureSkipVerify` feels a bit suspect. TODO?

Reply:
Hmm, commenting it out doesn't make any difference on the ckit tests, they're still passing with h2c. I'm taking a look as to what this _actually_ does and if we need it.

Reply:
And when _using_ TLS, the certificates should be able to match correctly, right? So I think we might be able to remove this

Reply:
> And when using TLS, the certificates should be able to match correctly, right? So I think we might be able to remove this

Yeah, we should never set this by default when using TLS unless the user specifies it.